network/p2p/unicast/manager.go

```diff
@@ -113,22 +200,41 @@ func (m *Manager) CreateStream(ctx context.Context, peerID peer.ID, maxAttempts
 //
 // Note that in case an existing TCP connection underneath to `peerID` exists, that connection is utilized for creating a new stream.
 // The multiaddr.Multiaddr return value represents the addresses of `peerID` we dial while trying to create a stream to it.
+// Expected errors during normal operations:
```

Comment:
From this method perspective, it is not clear that `peerID` is valid. So from the perspective of this method, this error condition is not necessarily an indicator of a bug or state corruption: https://github.com/onflow/flow-go/blob/77190f4772df8d06f24e3e29b6133208ad111c60/network/p2p/unicast/manager.go#L249-L252 it could just be that the calling code fed it with an invalid peer ID. Therefore, I think it is totally reasonable to 
1. represent this error case of `failed to dial remote peer` as a sentinel. 
2.  document that `swarm.ErrGaterDisallowedConnection` is a possible return value. Ideally we would wrap, `ErrGaterDisallowedConnection` so we don't expose libP2P internal errors to our business logic.

The calling code can always decide that it doesn't expect this sentinel.

Reply:
https://github.com/onflow/flow-go/pull/3902/commits/836d0b6898b147859309c965ec60b308e2720515